patsy/desc.py

```diff
@@ -593,8 +592,7 @@ def _assert_terms_match(terms, expected_intercept, expecteds, eval_env): # pragm
         if isinstance(term, Term):
             if isinstance(expected, str):
                 expected = (expected,)
-            assert term.factors == tuple([EvalFactor(s, eval_env)
-                                          for s in expected])
+            assert term.factors == tuple([EvalFactor(s) for s in expected])
```

Comment:
I think `eval_env` can probably be removed more thoroughly from this code? (I.e., if there aren't any other users in the function, then the variable can be removed entirely, and so on up the call stack.)